### PR TITLE
Upgrade to Python 3.9

### PR DESCRIPTION
**Pull request by 1st1**

### CODE REVIEW

**Comment by 1st1** [comment]
@ambv looks like everything is fine, except some deep magic in `edb/common/checked.py`. Probably some implementation details of `typing.py` changed in 3.9.


```
=====================================================================================================================================================================================
ERROR: test_checked (unittest.loader._FailedTest)
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
ImportError: Failed to import test module: test_checked
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/Users/yury/dev/edge/edgedb/tests/common/test_checked.py", line 415, in <module>
    class GenericFrozenCheckedSetSubclass(FrozenCheckedSet[T]):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/abc.py", line 85, in __new__
    cls = super().__new__(mcls, name, bases, namespace, **kwargs)
  File "/Users/yury/dev/edge/edgedb/edb/common/parametric.py", line 56, in __init_subclass__
    cls._init_parametric_user()
  File "/Users/yury/dev/edge/edgedb/edb/common/parametric.py", line 128, in _init_parametric_user
    raise TypeError(
TypeError: GenericFrozenCheckedSetSubclass: missing one or more type arguments for base 'FrozenCheckedSet'
```

Could you please take a look?

File: edb/common/_typing_inspect.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
+#
+# This source file is part of the EdgeDB open source project.
+#
+# Copyright 2011-present MagicStack Inc. and the EdgeDB authors.
```
Comment (elprans) [inline edb/common/_typing_inspect.py:4]:
```suggestion
# Copyright 2021-present MagicStack Inc. and the EdgeDB authors.
```

File: edb/common/_typing_inspect.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+# Portions copyright 2017-2020 Ivan Levkivskyi
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
```
Comment (elprans) [inline edb/common/_typing_inspect.py:11]: I think `typing_inspect` is released under MIT, and so this file should as well.

**Comment by elprans** [comment]
> ERROR: MarkupSafe-1.1.1-cp39-cp39-manylinux2010_x86_64.whl is not a supported wheel on this platform.

Let's slap `--no-binary` in CI to work this around.

File: edb/common/_typing_inspect.py (hunk 2 above)
Comment (1st1) [reply edb/common/_typing_inspect.py:11]: I'll fix

**Comment by 1st1** [comment]
> Let's slap --no-binary in CI to work this around.

For all packages?

**Comment by elprans** [comment]
No, I think only `MarkupSafe` is broken.

**Comment by elprans** [comment]
The actual error is because `python-version` hasn't been updated consistently in the workflow file, resulting in an attempt to install 3.9 wheels on 3.8.

**Comment by 1st1** [comment]
> The actual error is because python-version hasn't been updated consistently in the workflow file, resulting in an attempt to install 3.9 wheels on 3.8.

Yep, fixing that.

**Pull request merged**